daceml/autodiff/backward_pass_generator.py

```diff
@@ -800,7 +828,7 @@ def _add_gradient_data_descriptor(self, data_name: str):
 
         array = self.sdfg.arrays[data_name]
 
-        if type(array) not in [dt.Scalar, dt.Array, dt.View]:
+        if not isinstance(array, (dt.Scalar, dt.Array, dt.View)):
```

Comment:
this means that References are supported for autodiff as well. This is good, they should be.